exchange/bidder.go

```diff
@@ -231,7 +244,10 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 		}
 
 		if httpInfo.err == nil {
-			startTime := time.Now()
+			auctionOptions.response.mu.Lock()
+			auctionOptions.response.bidderResponseStartTime = time.Now()
```

Comment:
it doesn't appear that you do anything with `bidderResponseStartTime` in this PR. I assume that it exists for a later PR to use?

Reply:
yes will be used in later PR to calculate `PBS_auction_response` metric.

Reply:
@VeronikaSolovei9 

>It is not clear to me, please explain, why do you use mutex around time.Now()?
You don't modify anything between Lock and Unlock (for instance increment a counter or modify a struct in a concurrent mode). Per my knowledge this kind of operations are CPU "heavy" and we need to avoid them where it's possible. Reading time.Now() doesn't look like a reasonable candidate to be an atomic operation.

Mutex is needed because each go routine is trying to update `auctionOptions.response.bidderResponseStartTime`.  Here `auctionOptions` is being initialised by parent caller `getAllBids` and `auctionOptions.response.bidderResponseStartTime` is updated by go routines spawned from `getAllBids`


>Also if you want to save bidderResponseStartTime you need to write this value once. With current implementation every next bidder response will overwrite value in auctionOptions.response. Unless this is what you want to do.

We need to capture `bidderResponseStartTime` for the last bidder response. so we are overwriting it.

Reply:
>Mutex is needed because each go routine is trying to update auctionOptions.response.bidderResponseStartTime.

and

>Here auctionOptions.Response is being initialised once by caller getAllBids

Right, you create `auctionOptions.response.bidderResponseStartTime` at the "getAllBids" level, meaning this objects exists per auction request. This request may send requests to different bidders.
`bidderResponseStartTime` will be updated every time we read response from `httpInfo := <-responseChannel` and at the end it will have a time when the last valid response came.  Think about it this way: even if we run `bidder.doRequest` in parallel, responses will be read sequentially from `<-responseChannel` in order they appear there.
If you want to capture a time when last response came - it will work without it, because this object is instantiated per auction request (not per single bidder request-response). 
And you also said 
> We need to capture bidderResponseStartTime for the last bidder response. so we are overwriting it.

so yes, this is what it does and what it will do if you remove mutex lock and unlock. 

Name is also what is confusing here. Maybe rename it to "lastBidderResponseStartTime"? 

Just to explain the way I think about it. For a variable with name `bidderResponseStartTime` I would expect time will be written to this variable once for the very first response only, something like: 
```go
auctionOptions.response.mu.Lock()
if auctionOptions.response.bidderResponseStartTime == nil{
        auctionOptions.response.bidderResponseStartTime = time.Now()	
}
auctionOptions.response.mu.Unlock()	
```
And then mutex would make sense here.

Hope it makes sense.

Reply:
Renaming to `lastBidderResponseStartTime` sounds good to me. Also would like to know what opinion @hhhjort @mansinahar  have about this rename change

Reply:
@hhhjort  @VeronikaSolovei9  updated PR remove usage  of mutex.